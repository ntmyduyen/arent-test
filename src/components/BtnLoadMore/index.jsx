import Wrap from "./index.styles"

const BtnLoadMore = ({...props }) => {
    return (
        <Wrap {...props}>記録をもっと見る</Wrap>
    )
}
export default BtnLoadMore