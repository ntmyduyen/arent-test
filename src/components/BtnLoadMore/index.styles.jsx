import styled from 'styled-components'

const Wrap = styled.button`
    margin: 24px auto 0;
    border: none;
    outline: none;
    cursor: pointer;
    min-width: 296px;
    height: 56px;
    display: flex;
    align-items: center;
    justify-content: center;
    background: linear-gradient(32.95deg, var(--gradient-start) 8.75%, var( --gradient-end) 86.64%);

    font-style: normal;
    font-weight: 300;
    font-size: 18px;
    line-height: 26px;
    color: var(--color-light);
`;

export default Wrap