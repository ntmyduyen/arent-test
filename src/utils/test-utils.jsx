/* eslint-disable import/export */
import setupStore from '@redux/index';
import { cleanup, render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { afterEach } from 'vitest'

afterEach(() => {
	cleanup()
})

function renderWithProviders(
	ui,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = setupStore(preloadedState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>
			<MemoryRouter>{children}</MemoryRouter>
		</Provider>
	}
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
export {
	renderWithProviders
}