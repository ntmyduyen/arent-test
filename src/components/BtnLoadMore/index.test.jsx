import { renderWithProviders } from "@utils/test-utils"
import BtnLoadMore from "."

describe('BtnLoadMore', () => { 
    it('should be match with snapshot', () => {
        const { container } = renderWithProviders(<BtnLoadMore />)
        expect(container).toMatchSnapshot()
    })
})