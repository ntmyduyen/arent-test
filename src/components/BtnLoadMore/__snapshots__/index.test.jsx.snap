// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BtnLoadMore > should be match with snapshot 1`] = `
<div>
  <button
    class="sc-beqWaB exAsPU"
  >
    記録をもっと見る
  </button>
</div>
`;
